packages/react/src/Card/Card.tsx

```diff
@@ -0,0 +1,142 @@
+import React, {forwardRef, useCallback} from 'react'
+import {isFragment} from 'react-is'
+import clsx from 'clsx'
+
+import {Heading, HeadingTags, Text} from '..'
+import {ExpandableArrow} from '../ExpandableArrow'
+import type {BaseProps} from '../component-helpers'
+
+/**
+ * Design tokens
+ */
+import '@primer/brand-primitives/lib/design-tokens/css/tokens/functional/components/card/base.css'
+import '@primer/brand-primitives/lib/design-tokens/css/tokens/functional/components/card/colors-with-modes.css'
+
+/**
+ * Main stylesheet (as a CSS Module)
+ */
+import styles from './Card.module.css'
+import stylesLink from '../Link/Link.module.css'
+
+export type CardProps = {
+  /**
+   * Valid children include Card.Image, Card.Heading, and Card.Description
+   */
+  children: React.ReactNode | React.ReactElement<CardHeadingProps> | React.ReactElement<CardDescriptionProps>
+  /**
+   * The href of the link
+   * */
+  href: string
+  /**
+   * Changes the link label of the card
+   * * @default 'Learn more'
+   * */
+  label?: string
```

Comment:
@josepmartins - with the new requirements for supporting Label component in here I'm not wondering if `label` could be confusing. One option could be to replace `label` with `linkText` (which is what I think you did before? 😅). Then make Card.Label a separate child component. What do you think?